stdlib/src/bit/bit.mojo

```diff
@@ -145,7 +157,31 @@ fn bit_reverse[
 # ===----------------------------------------------------------------------===#
 # byte_swap
 # ===----------------------------------------------------------------------===#
-# TODO: implement byte_swap for Int type
+
+
+@always_inline("nodebug")
+fn byte_swap(val: Int) -> Int:
+    """Byte-swaps an integer value with an even number of bytes.
+
+    Byte swap an integer value with an even number of bytes (positive multiple
+    of 16 bits). This is equivalent to `llvm.bswap` intrinsic that has the
+    following semantics:
+
+    The `llvm.bswap.i16` intrinsic returns an i16 value that has the high and
+    low byte of the input i16 swapped. Similarly, the `llvm.bswap.i32` intrinsic
+    returns an i32 value that has the four bytes of the input i32 swapped, so
+    that if the input bytes are numbered 0, 1, 2, 3 then the returned i32 will
+    have its bytes in 3, 2, 1, 0 order. The `llvm.bswap.i48`, `llvm.bswap.i64`
+    and other intrinsics extend this concept to additional even-byte lengths (6
+    bytes, 8 bytes and more, respectively).
```

Comment:
Could you please update this with an explanation that doesn't refer to the llvm intrinsics? It would be good to fix this in the other `byte_swap` overload as well.

Reply:
just updated both explanation as well. Can you please take a look and make sure it looks good?